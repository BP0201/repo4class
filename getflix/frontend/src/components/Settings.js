import React, { useContext } from 'react';
import UserContext from '../UserContext';

import { Link } from 'react-router-dom';

const Settings = ({ deleteUser, logout }) => {
    const { currentUser } = useContext(UserContext);

    async function handleDeleteUser() {
        console.log("Deleting account")
        // remove user from db
        await deleteUser(currentUser.username);
        // log out user
        logout();
    }

    return (
        <div className='Settings container'>
            <h1 className='display-4 m-2'>Settings</h1>
            <p className='m-2'>Logged in as: {currentUser.username}</p>
            <ul className='list-unstyled'>
                <li className='border p-3'>
                    <Link to={`/edit/${currentUser.username}`} className='text-primary text-decoration-none'>
                        Change password
                    </Link>
                </li>
                <li onClick={handleDeleteUser} className='border p-3'>
                    <Link to="/" className='text-danger text-decoration-none'>
                        Delete account
                    </Link>
                </li>
            </ul>
        </div>
    )
}

export default Settings;