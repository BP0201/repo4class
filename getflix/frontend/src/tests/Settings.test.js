import React from "react";
import { render, fireEvent } from '@testing-library/react';
import Settings from '../components/Settings';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from './testUtils';

describe("Settings", () => {
  it("matches snapshot", () => {
    const { asFragment, getByText } = render(
      <MemoryRouter>
        <UserProvider>
          <Settings />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText("Logged in as: testUser")).toBeInTheDocument();
  });
});