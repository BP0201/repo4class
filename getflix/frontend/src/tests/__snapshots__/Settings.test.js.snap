// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Settings container"
  >
    <h1
      class="display-4 m-2"
    >
      Settings
    </h1>
    <p
      class="m-2"
    >
      Logged in as: testUser
    </p>
    <ul
      class="list-unstyled"
    >
      <li
        class="border p-3"
      >
        <a
          class="text-primary text-decoration-none"
          href="/edit/testUser"
        >
          Change password
        </a>
      </li>
      <li
        class="border p-3"
      >
        <a
          class="text-danger text-decoration-none"
          href="/"
        >
          Delete account
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
